### PR TITLE
Add PHP 7.4 & 8.0 support

### PR DESCRIPTION
**Pull request by Slamdunk**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -4,12 +4,11 @@ php:
   - 7.1
   - 7.2
   - 7.3
-  - 7.4snapshot
+  - 7.4
   - nightly
```
Comment (Ocramius) [inline .travis.yml:8]: Should be `8.0`, rather than `nightly`

File: classes/PHPTAL/FileSource.php [hunk 2]
```diff
@@ -40,7 +40,11 @@ public function getLastModifiedTime()
 
     public function getData()
     {
-        $content = file_get_contents($this->_path);
+        try {
+            $content = file_get_contents($this->_path);
+        } catch (Throwable $throwable) {
```
Comment (Ocramius) [inline classes/PHPTAL/FileSource.php:45]: Wait, since when does this throw? :O

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: .travis.yml (hunk 1 above)
Comment (Slamdunk) [reply .travis.yml:8]: This is a draft, there's a long journey before your review is requested ^^

Comment (Ocramius) [reply .travis.yml:8]: S'alright, just processed it as it hit my inbox :)

Comment (Slamdunk) [reply .travis.yml:8]: This phrase sounded a lot less harsh in my mind, sorry

File: .travis.yml [hunk 3]
```diff
@@ -4,23 +4,18 @@ php:
...
-  allow_failures:
-    - php: 7.4snapshot
-    - php: nightly
+  - 7.4
+  - 8.0
 
 before_script:
   # Make sure the locales used in the tests are installed
-  - composer install
+  - composer update --ignore-platform-req=php
```
Comment (Slamdunk) [inline .travis.yml:12]: I admin I was lazy and didn't bothered setting `--ignore-platform-req=php` only on PHP 8 cause of the state of this library, shame on me. Ping me if you want me to be more precise

File: classes/PHPTAL.php [hunk 4]
```diff
@@ -1223,7 +1223,7 @@ final public static function autoloadRegister()
         // Prepending PHPTAL's autoloader helps if there are other autoloaders
         // that throw/die when file is not found. Only >5.3 though.
         if (version_compare(PHP_VERSION, '5.3', '>=')) {
-            spl_autoload_register(array(__CLASS__,'autoload'), false, true);
+            @spl_autoload_register(array(__CLASS__,'autoload'), false, true);
```
Comment (Slamdunk) [inline classes/PHPTAL.php:1226]: This addresses a PHP 8.0 notice:

```
PHP Notice:  spl_autoload_register(): Argument #2 ($do_throw) has been ignored, spl_autoload_register() will always throw in ./PHPTAL/classes/PHPTAL.php on line 1226                                                  
```

File: phpunit.xml [hunk 5]
```diff
@@ -15,7 +15,10 @@ ensuring that your copy of PHPTAL works correctly.
     convertErrorsToExceptions="true"
     convertNoticesToExceptions="true"
     convertWarningsToExceptions="true">
-    <php><ini name="display_errors" value="1"/></php>
+    <php>
+        <ini name="display_errors" value="1"/>
+        <ini name="zend.exception_ignore_args" value="0"/>
```
Comment (Slamdunk) [inline phpunit.xml:20]: See https://www.php.net/manual/en/migration74.other-changes.php#migration74.other-changes.ini

File: tests/TalRepeatTest.php [hunk 6]
```diff
@@ -143,8 +143,8 @@ function testTraversableRepeat()
         $doc->loadXML('<a><b/><c/><d/><e/><f/><g/></a>');
 
         $tpl = $this->newPHPTAL();
-        $tpl->setSource('<tal:block tal:repeat="node nodes"><tal:block tal:condition="php:repeat.node.index==4">(len=${repeat/node/length})</tal:block>${repeat/node/key}${node/tagName}</tal:block>');
-        $tpl->nodes = $doc->getElementsByTagName('*');
+        $tpl->setSource('<tal:block tal:repeat="node nodes"><tal:block tal:condition="php:repeat.node.index==4">(len=${nodes/length})</tal:block>${repeat/node/key}${node/tagName}</tal:block>');
```
Comment (Slamdunk) [inline tests/TalRepeatTest.php:146]: That's a bug in the previous implementation, fixed in PHP >=8.0, `length` is only in the root node

File: tests/TalesRegistryTest.php [hunk 7]
```diff
@@ -42,12 +42,11 @@ public function prepareTemplate( Text_Template $template ) {
     }
 
     /**
-     * @runInSeparateProcess
```
Comment (Slamdunk) [inline tests/TalesRegistryTest.php:45]: I was unable to get this passing on a separate process, but it's ok in the main one now

File: classes/PHPTAL/FileSource.php (hunk 2 above)
Comment (Slamdunk) [reply classes/PHPTAL/FileSource.php:45]: 🤷
```
1) PhptalTest::testDirAsTemplate
file_get_contents(): read of 8192 bytes failed with errno=21 Is a directory

./PHPTAL/classes/PHPTAL/FileSource.php:43
./PHPTAL/classes/PHPTAL.php:1101
./PHPTAL/classes/PHPTAL.php:826
./PHPTAL/classes/PHPTAL.php:659
./PHPTAL/tests/PhptalTest.php:197
```

Comment (Ocramius) [reply classes/PHPTAL/FileSource.php:45]: I've seen this multiple times in other components that were upgraded to PHP 8

Interestingly in other libs I happen to maintain :| 

https://github.com/laminas/laminas-mail/issues/7

**Comment by Slamdunk** [comment]
Ping @Ocramius

**Review by Potherca — APPROVED** [review]

**Comment by Slamdunk** [comment]
@Potherca may I ask you how can I speed up the merge&release of this?

**Comment by Ocramius** [comment]
Hey @Slamdunk,

I've generally missed out on most OSS since January - will hopefully do a
release today, between meetings.

**Comment by Slamdunk** [comment]
Ping

**Comment by Ocramius** [comment]
I'll need to add release automation here too, before moving forward with a release @_@

**Comment by Ocramius** [comment]
Thanks @Slamdunk!

**Pull request merged**